pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,139 @@
+package localworkflows
+
+import (
+	"encoding/json"
+	"slices"
+	"strings"
+
+	"github.com/snyk/error-catalog-golang-public/snyk_errors"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/content_type"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
+	"github.com/snyk/go-application-framework/pkg/utils"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	FilterFindingsWorkflowName = "findings.filter"
+)
+
+var WORKFLOWID_FILTER_FINDINGS = workflow.NewWorkflowIdentifier(FilterFindingsWorkflowName)
+
+func InitFilterFindingsWorkflow(engine workflow.Engine) error {
+	flags := pflag.NewFlagSet(FilterFindingsWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_FILTER_FINDINGS, workflow.ConfigurationOptionsFromFlagset(flags), filterFindingsEntryPoint)
+
+	return err
+}
+
+type FindingsFilterFunc func(local_models.FindingResource) bool
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func filterFindingsBySeverity(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return utils.Contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func filterFindings(findingsModel local_models.LocalFinding, filter []FindingsFilterFunc) local_models.LocalFinding {
+	filteredFindings := local_models.LocalFinding{}
+	for _, finding := range findingsModel.Findings {
+		match := true
+		for _, filter := range filter {
+			if match {
```

Comment:
Suggestion: This logic seems like it can be improved a bit in order to be more clear and optimize performance. 

I think what we are looking to achieve is for each finding to see if any filter matches. With this assumption, we can just generally stop looping through filters if the first match is found.

Reply:
Not quite, we append a finding if it matches all the filters.
e.g. we can have two filters, `isBelowThreshold` and `isNotIgnored`, and to show the finding, it should match both

But yes, a small optimization can be made on the else branch 👍

Reply:
Oh, this intention wasn't clear to me. Can we please write this in a comment for `applyFilter` that mentions that multiple filters are ANDed. That would be really helpful.